subsys/bluetooth/host/hci_core.c

```diff
@@ -4066,6 +4082,7 @@ int bt_send(struct net_buf *buf)
 #if DT_HAS_CHOSEN(zephyr_bt_hci)
 	return bt_hci_send(bt_dev.hci, buf);
 #else
+	bt_buf_set_type(buf, bt_buf_out_type_from_h4_type(net_buf_pull_u8(buf)));
 	return bt_dev.drv->send(buf);
```

Comment:
> Does the BT monitor rely on CONFIG_BT_H4 being enabled?

It does not.

> If the BT monitor does not rely on H4, should we rely on the H4 type for that?

I may have failed at explaining the goal of this PR and the rest of the epic. I'll try differently:

This change has nothing to do with H4 besides reusing the header of that protocol. By header, I mean the "HCI packet indicator" (Core 4.A.2) aka. H4 type.

What are we reusing it for? In this PR, it's used as the *internal representation* in the Host. In follow-up PRs it will be used for the Host to driver interface. It replaces the use of bt_buf_get_type, which we want to get rid of because it uses the user_data field of the net buf to send data between layers. The h4_type will also automatically be part of any BT_LOG_HEXDUMP since it's just a part of the buffer, which is good for debugging.

The H4 and SPI drivers will benefit because the internal format will correspond with their on-wire formats, and this removes a conversion step. But this is not the primary motivator for this change.

Reply:
> We have 3 return paths, but only 1 of pulls the type from buf, and then we have an additional path with the monitor where the h4_type is also used.

I agree, it's clunky. But consider that the pull is only done on the deprecated path with to deal with the legacy `bt_dev.drv->send` API. The non-deprecated data paths have all been upgraded to accept the H4 typed buffers.

Reply:
Since the future steps are not immediately visible here: 
Will this function get further refactored in those future steps? If so, then I'm OK with this temporary version of it, but I wouldn't be happy about this version being there for a long period. 

> > Does the BT monitor rely on CONFIG_BT_H4 being enabled?
> 
> It does not.

> This change has nothing to do with H4 besides reusing the header of that protocol. By header, I mean the "HCI packet indicator" (Core 4.A.2) aka. H4 type.

But it relies on the type being an H4 type? I might be missing some details here, but can we always extract and use H4 types for the BT monitor even if we are not using a H4 driver? 
I assume you know what you are doing (you usually do), but it just seems a bit shady to extract a H4 type here and then use that as a `HCI packet indicator`. Are those always 1:1? 

Should the extracted variable that holds the type be renamed to not use `h4`?